### DISCUSSION TITLE
Why did you choose MedusaJS over other headless e-commerce solutions?

### DISCUSSION THREAD

**Discussion by genox** [medusajs/medusa#5136] [Q&A]
Just curious - 

For myself, I already developed a store with Saleor and wanted to use their 49 per month plan which they decided to drop 2 weeks ago, a month before I was supposed to launch. Then tried to get the OSS version running but it is too complicated, too many moving parts for entry level stores (their "plugins" essentially are autonomous app instances to be deployed as such, either docker or otherwise). Works for enterprise and an experienced devops engineer, maybe, but small stores are just going to be unmanageable due to the complexity and total surface of potential errors and misconfigurations.

After this "rug pull", self-hosting and OSS became my top requirements. I have been burned too many times in the past 10 years by business strategy switcharoos of SaaS offerings aiming at enterprise because "that's where the money is". I had a look at SaaS only offerings and there are a few I would use but I think I am mostly done with SaaS.. 

In the end, medusaJS is, while it has some disadvantages in the UI department compared to the other solutions I tested and could need something like a GraphQL API or a stable openAPI to generate typed clients, one of the easiest deployments and is extendable. Also since it is JS I feel more comfortable than a PHP or Python project.

As a plea to the fine folks here running this operation: Please always put OSS first and monetise in a way that doesn't make small ops unviable or create an enterprise behemoth. I would even consider paying regularly for premium support if it is within a certain budget, just to support the project.

The simplicity is one of the things making people flock here from other large solutions. Simplicity, OSS and self-hosting.

**Comment by timuric** [comment]
Hi @genox , Saleor team member here 👋

I understand the convenience plugins and medusa brings to a JS engineer, and it is great that you find it simple! 

>Works for enterprise and an experienced devops engineer, maybe, but small stores are just going to be unmanageable due to the complexity and total surface of potential errors and misconfigurations.

The opposite is true for plugins; instead of deploying a standalone app, you would have to deploy the whole commerce platform, ensuring nothing breaks, and your storefront remains operational and compatible with the changes. I get the argument that apps require standalone deployment, but so is a headless storefront, so if you are using something like Next.js with webhooks and API you can directly interact with Saleor using server components or functions. This means you would have to deploy a single target instead of two and not bother about keeping them compatible. The downside is that for some complex apps, you would need your own database, but for small projects, it can be trivial to use serverless solutions like Supabase, Vercel, etc.

> Please always put OSS first and monetise in a way that doesn't make small ops unviable or create an enterprise behemoth.

APIs and webhooks (even driven extensibility) are not a monetization strategy for enterprises. Millions of small businesses rely on APIs like Stripe, Shopify, etc. 

We always put open source first; we don't provide exclusivity to our cloud users or have any restrictions on the license. There are millions of developers are uncomfortable working with Node.JS or even JS developers who want to use different runtimes stacks prefer API based approach, so we believe this is inclusive and beneficial for the community.

I am not here to change anyone's opinion on Medusa, and I cheer for fellow open-source developers, but I felt this post needed some clarification for future readers.

**Reply by genox** [reply]
Not sure what your clarification is targeting other than to mention that the same goal can be achieved with various tools.

Also, guess how it feels to be faced with a massive price increase for no apparent reason right before launch.

Nobody said "sorry" for that, no "clarification" given for that.

So I guess I can be disappointed and I have the right to voice my opinion, which is, an opinion.

**Reply by timuric** [reply]
> Not sure what your clarification is targeting other than to mention that the same goal can be achieved with various tools.

That was my only intention! I don't question the right to be disappointed about our cloud pricing :)

**Comment by coopbri** [comment]
Hey! I went on a similar path as you -- from Saleor to Medusa. I eventually settled on [Vendure](https://vendure.io/), which is API-first and has native GraphQL support, with a very friendly community. I was looking for the exact same thing: native GraphQL or at least a nice OpenAPI story (from which GraphQL could be generated) -- Medusa has OpenAPI "support" but it's not super friendly to use in my experience. I guarantee it will improve over time though. But overall, I ended up on Vendure because it ticked all my boxes. There are a handful of things Medusa does better than Vendure and vice versa, but they both have nice plugin support so plenty of escape hatches/DIY ability. IMO, can't go wrong with either Medusa or Vendure.

One issue with Vendure is that the storefronts are lacking and the default admin UI is not very good in my opinion -- things Medusa does very well. I cared more about API-first design so this was not a dealbreaker for me and will naturally improve over time. Anyways, I agree wholeheartedly that simplicity, OSS, and self-hosting support are critical for any serious software adoption.

**Reply by genox** [reply]
I was on the fence about switching my stack a third time - to vendure, actually. A mix of .. lets call it .. "sunken cost", release pressure and just general unwillingness to rewrite my frontend for the third time in half a year made me stick with medusa. All the negative points I listed are still valid today but I am still thankful to have a solution and for the effort that has been put into medusajs so far.

Alas, it works and it works well for what it is used for. Bugs can be fixed, trust is hard to gain back.

**Reply by coopbri** [reply]
Very well said.
